### PR TITLE
fix: bugs found for timed exams

### PR DESCRIPTION
**Pull request by alangsto**
**JIRA:** [MST-1797](https://2u-internal.atlassian.net/browse/MST-1797)

**Description:** This PR adds a few fixes for issues I found while testing my frontend changes for https://github.com/openedx/frontend-lib-special-exams/pull/88. 

I also included a settings update to allow devstack to communicate with the exams service. This same wildcard is found in `production.py` so there shouldn't be an issue with adding it to `local.py`. 

**Merge checklist:**
- [ ] All reviewers approved
- [ ] CI build is green
- [ ] Changelog record added
- [ ] Documentation updated (not only docstrings)
- [ ] Commits are squashed

**Post merge:**
- [ ] Delete working branch (if not needed anymore)

### CODE REVIEW

File: edx_exams/apps/api/v1/views.py [hunk 1]
```diff
@@ -565,6 +565,8 @@ class CourseExamAttemptView(ExamsAPIView):
             }
     """
 
+    authentication_classes = (JwtAuthentication,)
```
Comment (Zacharis278) [inline edx_exams/apps/api/v1/views.py:568]: I think this need to be paired with a permission class (minimally IsAuthenticated) to work

File: edx_exams/apps/api/v1/views.py [hunk 2]
```diff
@@ -583,7 +589,8 @@ def get(self, request, course_id, content_id):
         serialized_exam['type'] = exam.exam_type
         serialized_exam['is_proctored'] = exam_type_class.is_proctored
         serialized_exam['is_practice_exam'] = exam_type_class.is_practice
-        serialized_exam['backend'] = exam.provider.verbose_name
+        # timed exams will have None as a backend
+        serialized_exam['backend'] = exam.provider.verbose_name if exam.provider is not None else None
```
Comment (varshamenon4) [inline edx_exams/apps/api/v1/views.py:593]: Nice catch!

File: edx_exams/apps/core/api.py [hunk 3]
```diff
@@ -233,7 +233,11 @@ def create_exam_attempt(exam_id, user_id):
     practice_exam_types = [PracticeExamType, OnboardingExamType]
 
     # if exam is past the due date, and it is a non-practice exam, raise error
-    if get_exam_type(exam_obj.exam_type) not in practice_exam_types and timezone.now() > exam_obj.due_date:
+    if (
```
Comment (varshamenon4) [inline edx_exams/apps/core/api.py:236]: Nice!

**Review by varshamenon4 — APPROVED** [review]
Nice!

**Pull request merged**